--- hmac-md5.c.orig	2011-05-21 09:36:49.076074063 +0200
+++ hmac-md5.c	2011-05-21 09:37:51.911556830 +0200
@@ -58,7 +58,7 @@
   md5_init_ctx (&inner);
 
   memset (block, IPAD, sizeof (block));
-  memxor (block, key, keylen);
+  gl_memxor (block, key, keylen);
 
   md5_process_block (block, 64, &inner);
   md5_process_bytes (in, inlen, &inner);
@@ -70,7 +70,7 @@
   md5_init_ctx (&outer);
 
   memset (block, OPAD, sizeof (block));
-  memxor (block, key, keylen);
+  gl_memxor (block, key, keylen);
 
   md5_process_block (block, 64, &outer);
   md5_process_bytes (innerhash, 16, &outer);
