--- version-etc.c.orig	2007-07-12 22:23:47.000000000 +0200
+++ version-etc.c	2007-07-12 22:24:01.000000000 +0200
@@ -67,7 +67,7 @@
 
   fputs (_("\
 \n\
-License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>\n\
+License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl.html>\n\
 This is free software: you are free to change and redistribute it.\n\
 There is NO WARRANTY, to the extent permitted by law.\n\
 \n\
