diff --git a/m4/year2038.m4 b/m4/year2038.m4
index 2e4427e6fa..dc80d25c79 100644
--- a/m4/year2038.m4
+++ b/m4/year2038.m4
@@ -15,13 +15,6 @@ m4_ifdef([AC_SYS_YEAR2038], [
 
 AC_DEFUN([gl_YEAR2038_EARLY],
 [
-  AC_REQUIRE([AC_CANONICAL_HOST])
-  case "$host_os" in
-    mingw*)
-      AC_DEFINE([__MINGW_USE_VC2005_COMPAT], [1],
-        [For 64-bit time_t on 32-bit mingw.])
-      ;;
-  esac
 ])
 
 # gl_YEAR2038_TEST_INCLUDES
