--- lgl/gc-libgcrypt.c.orig	2008-05-05 13:26:28.000000000 +0200
+++ lgl/gc-libgcrypt.c	2008-05-05 13:26:43.000000000 +0200
@@ -145,7 +145,7 @@
       gcryalg = GCRY_CIPHER_RFC2268_40;
       break;
 
-#ifdef HAVE_CAMELLIA
+#ifdef ENABLE_CAMELLIA
     case GC_CAMELLIA128:
       gcryalg = GCRY_CIPHER_CAMELLIA128;
       break;
@@ -294,9 +294,11 @@
       gcryalg = GCRY_MD_SHA512;
       break;
 
+#ifdef ENABLE_CAMELLIA
     case GC_SHA224:
       gcryalg = GCRY_MD_SHA224;
       break;
+#endif
 
     case GC_RMD160:
       gcryalg = GCRY_MD_RMD160;
@@ -397,9 +399,11 @@
       len = GC_SHA512_DIGEST_SIZE;
       break;
 
+#ifdef ENABLE_CAMELLIA
     case GC_SHA224:
       len = GC_SHA224_DIGEST_SIZE;
       break;
+#endif
 
     default:
       return 0;
