--- lgl/gc-libgcrypt.c.orig	2008-04-13 14:51:05.000000000 +0200
+++ lgl/gc-libgcrypt.c	2008-04-13 14:52:04.000000000 +0200
@@ -294,9 +294,11 @@
       gcryalg = GCRY_MD_SHA512;
       break;
 
+#ifdef ENABLE_CAMELLIA
     case GC_SHA224:
       gcryalg = GCRY_MD_SHA224;
       break;
+#endif
 
     case GC_RMD160:
       gcryalg = GCRY_MD_RMD160;
@@ -397,9 +399,11 @@
       len = GC_SHA512_DIGEST_SIZE;
       break;
 
+#ifdef ENABLE_CAMELLIA
     case GC_SHA224:
       len = GC_SHA224_DIGEST_SIZE;
       break;
+#endif
 
     default:
       return 0;
